functions/common.py

```diff
@@ -1,26 +1,108 @@
 import exceptions.exceptions as exceptions
 
 
-def factorial(num: int) -> int:
-    if not isinstance(num, int):
-        # Input should be an integer
-        raise exceptions.InputError(num, "Error: non-integer factorial")
-    elif num < 0:
-        # Input should be non-negative
-        raise exceptions.InputError(num, "Error: negative factorial")
-    elif num == 0:
-        return 1
+def factorial(argument: int) -> int:
+    """Returns the product of all non-negative integers less than or equal to some non-negative integer.
+
+    Args:
+        argument (int): Non-negative integer used as input to factorial function
+
+    Returns:
+        int: Product of all non-negative integers less than or equal to 'argument'
+        
+    Raises:
+        InputError: If 'argument' is not a non-negative integer
+    """
+    # Ensure that input is an integer.
+    if int(argument) != argument:
+        raise exceptions.InputError(argument, "Input to factorial should be a non-negative integer.")
+    argument = int(argument)
+    # Ensure that input is non-negative.
+    if argument < 0:
+        raise exceptions.InputError(argument, "Input to factorial should be a non-negative integer.")
+        
+    # Calculate factorial using repeated multiplication.
     result = 1
-    for i in range(1, num + 1):
+    for i in range(1, argument + 1):
         result *= i
     return result
 
 
-def mod(dividend: float, divisor: float) -> float:
```

Comment:
Why was mod function removed, just curious.

Reply:
I didn't see it being used and I figured we should use % instead anyways (unless the mod function does something different). Let me know if I should revert it though.